src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -346,28 +380,37 @@ export class DataSourceSavedObjectsClientWrapper {
             'Update failed due to deprecated data source: "auth.credentials.username" missing. Please delete and create another data source.'
           );
         }
+        return this.validateEncryptionContext(password, endpoint);
+      case AuthType.SigV4:
+        const { accessKey } = auth.credentials as SigV4Content;
+        return this.validateEncryptionContext(accessKey, endpoint);
```

Comment:
what encryption context we are using for sigv4?

Reply:
same as other types, we use `endpoint`

```
  case AuthType.UsernamePasswordType:
        // Signing the data source with endpoint
        return {
          ...attributes,
          auth: await this.encryptBasicAuthCredential(auth, { endpoint }),
        };
      case AuthType.SigV4:
        return {
          ...attributes,
          auth: await this.encryptSigV4Credential(auth, { endpoint }),
        };
```